framework/src/media/BufferOutputDataSource.cpp

```diff
@@ -0,0 +1,78 @@
+/* ****************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <stdio.h>
+#include <debug.h>
+#include <media/BufferOutputDataSource.h>
+
+namespace media {
+namespace stream {
+
+BufferOutputDataSource::BufferOutputDataSource(recorder_callback_t callback)
+	: OutputDataSource(), mCallback(callback)
+{
+}
+
+BufferOutputDataSource::BufferOutputDataSource(unsigned int channels, unsigned int sampleRate, audio_format_type_t pcmFormat, recorder_callback_t callback)
+	: OutputDataSource(channels, sampleRate, pcmFormat), mCallback(callback)
+{
+}
+
+BufferOutputDataSource::BufferOutputDataSource(const BufferOutputDataSource& source) :
+	OutputDataSource(source), mCallback(source.mCallback)
+{
+}
+
+BufferOutputDataSource& BufferOutputDataSource::operator=(const BufferOutputDataSource& source)
+{
+	OutputDataSource::operator=(source);
+	return *this;
+}
+
+bool BufferOutputDataSource::open()
+{
+	return true;
+}
+
+bool BufferOutputDataSource::close()
+{
+	return true;
+}
+
+bool BufferOutputDataSource::isPrepare()
+{
+	return true;
+}
+
+ssize_t BufferOutputDataSource::write(unsigned char* buf, size_t size)
+{
+	if (!buf) {
```

Comment:
please, add condition for size

Reply:
@jsdosa For example, are you talking about a negative value?

Reply:
yes, you should check the value.

Reply:
@jsdosa i fixed.